functions/src/tagApplicants/index.js

```diff
@@ -0,0 +1,90 @@
+const { db, functions } = require('../utils/firestore');
+
+const mapApplicantTypeToCollection = {
+  0: 'hacker_short_info',
+  1: 'volunteer_short_info',
+};
+
+const BATCH_LIMIT = 500;
+
+// cloud function for submitting hacker application
+exports.tagApplicants = functions.https.onRequest((request, response) => {
+  // set origin and headers to allow CORS for local testing
+  // https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
+  response.set('Access-Control-Allow-Origin', '*');
+  response.set('Access-Control-Allow-Headers', 'Content-Type, crossDomain');
+
+  // this function only allows POST requests
+  response.set('Access-Control-Allow-Methods', 'POST');
+  if (request.method !== 'POST') {
+    return response.status(400).send('Unrecognized method'); // database error
+  }
+
+  // ensure request.body is an object, not a string
+  const data = typeof request.body !== 'object' ? JSON.parse(request.body) : request.body;
+
+  const {
+    applicantType,
+    applicants,
+    tagsToAdd,
+    tagsToRemove,
+  } = data;
+
+  const collection = mapApplicantTypeToCollection[applicantType];
+  if (!collection) {
+    return response.status(400).send('Unrecognized applicant type');
+  }
+
+  if (!tagsToAdd || tagsToAdd.constructor !== Array) {
+    return response.status(400).send('tagsToAdd field is missing/not an array');
+  }
+
+  if (!tagsToRemove || tagsToRemove.constructor !== Array) {
+    return response.status(400).send('tagsToRemove field is missing/not an array');
+  }
+
+  if (!applicants || applicants.constructor !== Array) {
+    return response.status(400).send('applicants field is missing/not an array');
+  }
+
+  // eslint-disable-next-line max-len
+  return Promise.all(applicants.map(applicantId => db.collection(collection).doc(applicantId).get()))
```

Comment:
We should also check that the user who's calling this is an admin like in the backfill functions.